core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -575,8 +575,13 @@ class SparkSubmitSuite
   // NOTE: This is an expensive operation in terms of time (10 seconds+). Use sparingly.
   private def runSparkSubmit(args: Seq[String]): Unit = {
     val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))
+    val sparkSubmit = if (Utils.isWindows) {
+      Seq(new File("..\\bin\\spark-submit.cmd").getAbsolutePath) ++ args
```

Comment:
> The path here is different (starting with ..). Really the right way to do this is with `new File("./bin/spark-submit"), for non-Windows as well, optionally putting .cmd on the end for Windows.

@srowen I've verified this. On windows, the working directory would be spark/core here. So `./bin/spark-submit` doesn't work here on windows. Further, it seems we have to add the `cmd` suffix, otherwise it will complain that it's not a valid win32 program.